policy/modules/system/systemd.te

```diff
@@ -1527,8 +1541,17 @@ seutil_libselinux_linked(systemd_user_session_type)
 allow systemd_user_runtime_dir_t self:capability { fowner chown sys_admin dac_read_search dac_override };
 allow systemd_user_runtime_dir_t self:process setfscreate;
 
+list_dirs_pattern(systemd_user_runtime_dir_t, systemd_user_runtime_dir_t, systemd_user_runtime_dir_t)
+read_files_pattern(systemd_user_runtime_dir_t, systemd_user_runtime_dir_t, systemd_user_runtime_dir_t)
+
 domain_obj_id_change_exemption(systemd_user_runtime_dir_t)
 
+manage_dirs_pattern(systemd_user_runtime_dir_t, systemd_user_runtime_t, systemd_user_runtime_t)
+manage_files_pattern(systemd_user_runtime_dir_t, systemd_user_runtime_t, systemd_user_runtime_t)
+manage_sock_files_pattern(systemd_user_runtime_dir_t, systemd_user_runtime_t, systemd_user_runtime_t)
+manage_fifo_files_pattern(systemd_user_runtime_dir_t, systemd_user_runtime_t, systemd_user_runtime_t)
+manage_chr_files_pattern(systemd_user_runtime_dir_t, systemd_user_runtime_t, systemd_user_runtime_t)
+
```

Comment:
I guess you want this for backwards compatibility? the above should no longer be needed

Reply:
as per systemd v247 (i think) systemd iself maintains /run/user/*/systemd/inaccessible

Reply:
In my testing `systemd-user-runtime-dir` is what is unlinking these files:

```
time->Sat Mar 13 09:05:26 2021
type=PROCTITLE msg=audit(1615644326.971:3695): proctitle=2F6C69622F73797374656D642F73797374656D642D757365722D72756E74696D652D64697200
73746F700031303030
type=PATH msg=audit(1615644326.971:3695): item=1 name="chr" inode=8 dev=00:1f mode=020000 ouid=1000 ogid=1000 rdev=00:00 obj=staff_u:
object_r:systemd_user_runtime_t:s0 nametype=DELETE cap_fp=0 cap_fi=0 cap_fe=0 cap_fver=0 cap_frootid=0
type=PATH msg=audit(1615644326.971:3695): item=0 name="/" inode=3 dev=00:1f mode=040755 ouid=1000 ogid=1000 rdev=00:00 obj=staff_u:ob
ject_r:systemd_user_runtime_t:s0 nametype=PARENT cap_fp=0 cap_fi=0 cap_fe=0 cap_fver=0 cap_frootid=0
type=CWD msg=audit(1615644326.971:3695): cwd="/"
type=SYSCALL msg=audit(1615644326.971:3695): arch=c000003e syscall=263 success=no exit=-13 a0=6 a1=1554f003 a2=0 a3=76 items=2 ppid=1
 pid=112521 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="systemd-user-ru" 
exe="/lib/systemd/systemd-user-runtime-dir" subj=system_u:system_r:systemd_user_runtime_dir_t:s0 key=(null)
type=AVC msg=audit(1615644326.971:3695): avc:  denied  { unlink } for  pid=112521 comm="systemd-user-ru" name="chr" dev="tmpfs" ino=8
 scontext=system_u:system_r:systemd_user_runtime_dir_t:s0 tcontext=staff_u:object_r:systemd_user_runtime_t:s0 tclass=chr_file permiss
ive=0
```
Though if anything I think we can reduce these accesses down to just `delete_file_perms` and others.

Reply:
Looks like I can just drop these accesses and add an interface to delete `chr_file` like below.